app/code/Magento/Ups/Test/Unit/Model/CarrierTest.php

```diff
@@ -445,8 +445,25 @@ public function testRequestToShipment(array $requestData, array $expectedRequest
     {
         /** @var \Magento\Shipping\Model\Shipment\Request $request */
         $request = $this->helper->getObject(\Magento\Shipping\Model\Shipment\Request::class);
+        $shipmentMock = $this->getMockBuilder(\Magento\Sales\Model\Order\Shipment::class)
+            ->disableOriginalConstructor()
+            ->onlyMethods(['getOrder'])
+            ->getMock();
+        $orderMock =  $this->getMockBuilder(\Magento\Sales\Model\Order::class)
+            ->disableOriginalConstructor()
+            ->onlyMethods(['getIncrementId'])
+            ->getMock();
+
+        $shipmentMock->expects($this->any())
+            ->method('getOrder')
+            ->willReturn($orderMock);
+        $orderMock->expects($this->any())
+            ->method('getIncrementId')
+            ->willReturn('100000001');
+
+        $requestData['order_shipment'] = $shipmentMock;
         $request->setData($requestData);
-        $request->setPackages([['items' => []]]);
+        $request->setPackages([['items' => [], 'params' => ['container' => '']]]);
```

Comment:
Could you please check on a live Magento that those array keys always present and we will not hit same issue in runtime?
https://github.com/magento/magento2/blob/1807f70756b521240c06eb556802422307cabbb5/app/code/Magento/Ups/Model/Carrier.php#L1523-L1524

Reply:
Hello, @xmav  thanks for review, I checked on a live Magento that those array keys always presents, they are added to the request from the configuration, that's why we will not have this problem in runtime.

Thank you.